tmail-backend/tmail-third-party/open-ai/src/main/java/com/linagora/tmail/mailet/AIBotConfig.java

```diff
@@ -0,0 +1,114 @@
+package com.linagora.tmail.mailet;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import jakarta.mail.internet.AddressException;
+
+import org.apache.james.core.MailAddress;
+import org.apache.mailet.MailetConfig;
+import org.apache.mailet.MailetException;
+
+import com.google.common.base.Strings;
+
+public class AIBotConfig {
+    public static String API_KEY_PARAMETER_NAME = "apiKey";
+    public static String GPT_ADDRESS_PARAMETER_NAME = "gptAddress";
+    public static String MODEL_PARAMETER_NAME = "model";
+    public static String BASE_URL_PARAMETER_NAME = "baseURL";
+
+    public static final LlmModel DEFAULT_LLM_MODEL =
+            new LlmModel(Llm.OPEN_AI, "gpt-4o-mini");
+
+    private final String apiKey;
+    private final String baseURL;
+    private final MailAddress gptAddress;
+    private final LlmModel llmModel;
+
+    public AIBotConfig(String apiKey, String baseURL, MailAddress gptAddress, LlmModel llmModel) {
+        Objects.requireNonNull(apiKey);
+        Objects.requireNonNull(gptAddress);
+        Objects.requireNonNull(llmModel);
+
+        this.apiKey = apiKey;
+        this.baseURL = baseURL;
+        this.gptAddress = gptAddress;
+        this.llmModel = llmModel;
+    }
+
+    public static AIBotConfig fromMailetConfig(MailetConfig mailetConfig) throws MailetException {
+        String apiKeyParam = mailetConfig.getInitParameter(API_KEY_PARAMETER_NAME);
+        String gptAddressParam = mailetConfig.getInitParameter(GPT_ADDRESS_PARAMETER_NAME);
+        String llmModelParam = mailetConfig.getInitParameter(MODEL_PARAMETER_NAME);
+        String baseUrlParam = mailetConfig.getInitParameter(BASE_URL_PARAMETER_NAME);
+
+        if (Strings.isNullOrEmpty(apiKeyParam)) {
+            throw new MailetException("No value for " + API_KEY_PARAMETER_NAME + " parameter was provided.");
+        }
+
+        if (Strings.isNullOrEmpty(gptAddressParam)) {
+            throw new MailetException("No value for " + GPT_ADDRESS_PARAMETER_NAME + " parameter was provided.");
+        }
+        
+        MailAddress mailAddress;
+        try {
+            mailAddress = new MailAddress(gptAddressParam);
+        } catch (AddressException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Can we widen the exception thrown to avoid variable reallocation?

Reply:
Why is variable reallocation bad?

Reply:
i think it's just for code readability, it's better if a variable always refers to the same thing :)

Reply:
Yes here it is not that hard but it can quickly go out of control.

Reply:
Benoit can you explain in more details what you mean by avoid variable reallocation, what code changes can I make?

Reply:
Read the book "Clean code"

https://community.sonarsource.com/t/new-rule-variable-should-not-be-reassigned/78154

Also on communication avoid designating a single individual. Aren't other able to provide the explaination?

Reply:
(Also another great tip is to avoid 'you' and favor 'we' - this forces to think collective together and prevent antagonising people)